### ISSUE TITLE
Prepare extension for 0.19

### ISSUE THREAD

**Issue by hakonrossebo** [Krzysztof-Cieslak/vscode-elm#219]
While 0.19 discussions should be kept in the Elm Slack #elm-dev channel, we should coordinate efforts to prepare for Elm 0.19.

This extension relies on elm-oracle. Elm-oracle has not been updated for some time, but by looking on the source code, it seems like it should continue work successfully. For other linting/auto-completion, we should verify that all changes in 0.19 are covered.

One minor breaking change will be that there is only one elm binary for all command line-operations. We will need to change some calls to command-line operations.

Todo:

- [ ] Verify elm-oracle
- [ ] Verify user-projects, symbols code
- [ ] Elm-reactor operations
- [ ] Elm-repl operations
- [ ] Elm-make operations
- [ ] Elm-package operations

Please suggest any missing tasks here.

How should we proceed? Should we have a separate branch for 0.19 PR's?

**Comment by sentience** [comment]
Based on my testing, elm-oracle does not recognise the installed dependencies of an Elm 0.19 project. When 0.19 alpha’s `elm install` has been run for a project, `elm-oracle` simply outputs this error:

```
[{"error":"Dependencies file is missing. Perhaps you need to run `elm-package install`?"}]
```

Based on this, I think elm-oracle will need to be updated for Elm 0.19.

**Comment by hakonrossebo** [comment]
Based on feedback from Evan in the Elm-community Slack channel, upgrading elm-oracle could be difficult and unsupported. We should consider alternatives. A language server would be the best solution, but in the meantime, is there any other editor plugin implementations for parsing Elm that we could use?

**Comment by axelo** [comment]
Just for brevity I tried to emulate a 0.18 world for 0.19, worked some what. 

Se https://gist.github.com/axelo/793a652fdd4f88c8e60f76933b9899fd 

It may or may not work for others, but when trying out 0.19 always nice to not lose all tooling.

And to get errors to get reported correctly one can temporary patch elmLinter.js,

Se  https://gist.github.com/axelo/4446c12712e8f8b9356c3a79e84a9216 as POC.

**Comment by 7sharp9** [comment]
See #242 for a PR in progress, feel free to join.

**Comment by hakonrossebo** [comment]
@axelo - thanks for the error reporting json patch - I'll try to integrate it now.

**Comment by Krzysztof-Cieslak** [comment]
Fixed by #243
